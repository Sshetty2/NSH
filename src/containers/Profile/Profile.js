import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import EventContainer from '../EventContainer/EventContainer';
import { setUser } from '../../actions';
import { mockUser } from '../../data/mockData';

export const Profile = (props) => {
  const [contentEditable, toggleEditable] = useState(false);
  const buttonText = contentEditable ? 'Save' : 'Edit Profile';

  return (
    <div className="Profile">
      <div className="user-intro-section">
        <img
          src={mockUser.image_url}
          alt="user headshot"
          className="user-photo"
        />
        <h1 className="user-name" contentEditable={contentEditable}>{mockUser.name.toUpperCase()}</h1>
      </div>
      <div className="user-bio-section">
        <button
          onClick={() => toggleEditable(!contentEditable)}
          className="edit-profile-button"
        >
          {buttonText}
        </button>
        <p className="user-about-me">ABOUT ME</p>
        <p className="user-bio" contentEditable={contentEditable}>
          {mockUser.bio}
        </p>
      </div>
      <div className="view-tabs-section">
        <NavLink
          exact to='/profile/upcoming'
          className='upcoming-button'
          activeClassName='active'>
          Upcoming
        </NavLink>
        <NavLink
          exact to='/profile/wishlist'
          className='wishlist-button'
          activeClassName='active'>
          Wishlist
        </NavLink>
        <NavLink
          exact to='/profile/past'
          className='past-button'
          activeClassName='active'>
          Past
        </NavLink>
      </div>
      <EventContainer pathname={props.location.pathname}/>
    </div>
  );
};

export const mapDispatchToProps = dispatch => ({
  setUser: user => dispatch(setUser(user))
});

export default connect(
  null,
  mapDispatchToProps
)(Profile);

Profile.propTypes = {
  setUser: PropTypes.func
};
