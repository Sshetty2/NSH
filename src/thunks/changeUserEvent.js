import { setUserEvents, setLoading, setError } from '../actions';
import { getUserEvents } from './getUserEvents';
import { fetchData } from '../utils/api';

export const changeUserEvent = (eventId, method, status = '') => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const path = status.length ? `?status=${status}` : status;
      await fetchData(`/users/1/events/${eventId}` + path, method);
      const events = await getUserEvents('1');
      dispatch(setUserEvents(events.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false))
  }
}