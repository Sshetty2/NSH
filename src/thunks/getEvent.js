import { setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getEvent = (eventId = '') => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const event = await fetchData(`/events/${eventId}`, 'GET');
      dispatch(setLoading(false));
      return event.data.attributes;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}