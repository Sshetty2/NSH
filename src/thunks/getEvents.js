import { setEvents, setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getEvents = (limit = null, event_type = null, state = null, month = null) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      let path = (limit || event_type || state || month) ? '?' : '';
      if (limit) path = path + `&limit=${limit}`;
      if (event_type) path = path + `&event-type=${event_type}`;
      if (state) path = path + `&state=${state}`;
      if (month) path = path + `&month=${month}`;
      const events = await fetchData('/events' + path, 'GET');
      dispatch(setEvents(events.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false))
  }
}