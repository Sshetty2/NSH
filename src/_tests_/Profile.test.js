import React from 'react';
import { shallow } from 'enzyme';
import { Profile, mapStateToProps, mapDispatchToProps } from '../containers/Profile/Profile';
import { updateUser } from '../thunks/updateUser';
import { mockUser } from '../data/mockData';
jest.mock('../thunks/updateUser');

const mockProps = {
  updateUser: jest.fn(),
  user: mockUser,
  location: { pathname: '/profile' }
};

describe('Profile', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Profile {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot for loading', () => {
    const mockProps = {
      updateUser: jest.fn(),
      user: {},
      location: { pathname: '/profile' }
    };
    wrapper = shallow(<Profile {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return a props object with events', () => {
      const mockState = {
        user: { name: 'Wonderwoman', bio: 'About me'},
        isLoading: true,
        error: ''
      };
      const expectedState = {
        user: { name: 'Wonderwoman', bio: 'About me' },
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the updateUser action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = updateUser();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.updateUser();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
