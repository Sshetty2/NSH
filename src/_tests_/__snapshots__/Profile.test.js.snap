// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match the snapshot 1`] = `
<div
  className="Profile"
>
  <div
    className="user-intro-section"
  >
    <img
      alt="user headshot"
      className="user-photo"
      src="https://alumni.turing.io/sites/default/files/styles/graduate_full_profile/public/squareheadshot1.jpg?itok=EGlR-8YR"
    />
    <h1
      className="user-name"
    >
      JANE DOE
    </h1>
  </div>
  <div
    className="user-bio-section"
  >
    <p
      className="user-about-me"
    >
      About Me
    </p>
    <button
      className="edit-profile-button"
      onClick={[Function]}
    >
      Edit Profile
    </button>
    <p
      className="user-bio"
      contentEditable={false}
    >
      Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!! Hi I’m Jane and I have no friends because this app only has one user, and it’s me!!!
    </p>
  </div>
  <div
    className="view-tabs-section"
  >
    <button
      className="upcoming-button"
    >
      Upcoming
    </button>
    <button
      className="wishlist-button"
    >
      Wishlist
    </button>
    <button
      className="past-button"
    >
      Past
    </button>
  </div>
  <EventContainer />
</div>
`;
